src/Microsoft.Identity.Client/ApiConfig/AcquireTokenInteractiveParameterBuilder.cs

```diff
@@ -162,6 +166,11 @@ protected override void Validate()
             {
                 throw new InvalidOperationException(CoreErrorMessages.ActivityRequiredForParentObjectAndroid);
             }
+#elif iOS
+            if(_ownerWindow is UIViewController uiViewController)
+            {
+                Parameters.UiParent.SetUIViewController(uiViewController);
+            }
```

Comment:
>} [](start = 12, length = 1)

should there be an else statement throwing an exception like for Android if this is not set? Or can things work with-out this? e.g. only relevant for broker scenarios?

Reply:
I tested w/system browser and embedded webview and things work fine. we have a check for android above. we need a uiViewcontroller for broker. if we don't set it now, we don't have a viewController to check for broker later.

---
In reply to: [258168321](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/881#discussion_r258168321) [](ancestors = 258168321)